.github/workflows/ci.yml

```diff
@@ -0,0 +1,29 @@
+name: Python check
+
+on:
+  pull_request:
+
+jobs:
+  pre_commit_checks:
+    name: Pre-Commit checks
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@main
+      - uses: astral-sh/setup-uv@v5
+        with:
+          enable-cache: true
+
+      - name: Setup uv python environment
+        working-directory: ./functions/generated/pyfirebase
+        run: uv venv
+
+      - name: uv lock check
+        working-directory: ./functions/generated/pyfirebase
+        run: uv lock --locked --offline
+
+      - name: uv sync
+        working-directory: ./functions/generated/pyfirebase
+        run: uv sync --all-extras
+
+      - uses: pre-commit/action@v3.0.1
```

Comment:
We also need a check that identifies if the generated python files are up-to-date.

Reply:
I have added another issues for this task.